Lib/test/test_contextlib_async.py

```diff
@@ -536,6 +556,18 @@ async def suppress_exc(*exc_details):
         self.assertIsInstance(inner_exc, ValueError)
         self.assertIsInstance(inner_exc.__context__, ZeroDivisionError)
 
+    @_async_test
+    async def test_instance_bypass_async(self):
```

Comment:
What is this test doing?

Reply:
It is a copy of similar test for synchronous context managers (`test_instance_bypass`), but for asynchronous context managers. It tests that special methods are looked up in a class, not in an instance. So merely setting `__aenter__` and `__aexit__` as instance attributes does not make a context manager. `enter_async_context()` fails with relevant message, and `push_async_exit()` treats argument as a callable, not context manager.